### PR TITLE
[pipelines] external vault adoption

### PR DESCRIPTION
**Pull request by rahulnlb1**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [X] Variables and other changes are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
This PR is added to support external vault.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

File: stable/pipelines/README.md [hunk 1]
```diff
@@ -157,6 +156,23 @@ rabbitmq:
...
+global:
+  vault:
+    url: vault_url
+    token: vault_token
+```
+
+```bash
+helm upgrade --install pipelines --namespace pipelines center/jfrog/pipelines -f values-external-vault.yaml
+```
+
```
Comment (rimusz) [inline stable/pipelines/README.md:175]: please add external secret for Vault token example as well, as we have in Pipelines v1.4.4 chart

File: stable/pipelines/templates/pipelines-statefulset.yaml [hunk 2]
```diff
@@ -64,11 +64,9 @@ spec:
             echo "Waiting for Vault to come up...";
             {{- if .Values.vault.enabled }}
             until nc -z -w 2 {{ include "pipelines.vault.name" . }} {{ .Values.vault.service.port }} && echo Vault ok; do
-            {{- else }}
-            until nc -z -w 2 {{ tpl .Values.global.vault.host . }} {{ .Values.global.vault.port }} && echo Vault ok; do
-            {{- end }}
               sleep 2;
             done;
+            {{- end }}
```
Comment (rimusz) [inline stable/pipelines/templates/pipelines-statefulset.yaml:69]: please add external vault url uptime check

**Review by rimusz — CHANGES_REQUESTED** [review]

File: stable/pipelines/templates/pipelines-statefulset.yaml [hunk 3]
```diff
@@ -64,11 +64,25 @@ spec:
...
             {{- if .Values.vault.enabled }}
             until nc -z -w 2 {{ include "pipelines.vault.name" . }} {{ .Values.vault.service.port }} && echo Vault ok; do
+              sleep 2;
+            done;
             {{- else }}
-            until nc -z -w 2 {{ tpl .Values.global.vault.host . }} {{ .Values.global.vault.port }} && echo Vault ok; do
-            {{- end }}
+            is_vault_up=0;
+            initilized_check_endpoint="/v1/sys/init";
+            until [ "$is_vault_up" == "1" ] && echo Vault ok; do
```
Comment (danielezer) [inline stable/pipelines/templates/pipelines-statefulset.yaml:72]: Use two square brackets here. It is also better to use two instead of one.

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/pipelines/README.md (hunk 1 above)
Comment (rahulnlb1) [reply stable/pipelines/README.md:184]: Added the external secret example for vault

File: stable/pipelines/templates/pipelines-statefulset.yaml (hunk 2 above)
Comment (rahulnlb1) [reply stable/pipelines/templates/pipelines-statefulset.yaml:85]: Added the check for external vault availability.

File: stable/pipelines/templates/pipelines-statefulset.yaml (hunk 3 above)
Comment (rahulnlb1) [reply stable/pipelines/templates/pipelines-statefulset.yaml]: Used two square brackets

**Review by danielezer — APPROVED** [review]

**Comment by rahulnlb1** [comment]
recheckcla

**Comment by rahulnlb1** [comment]
I have read the CLA Document and I hereby sign the CLA

File: stable/pipelines/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # JFrog Pipelines Chart Changelog
 All changes to this chart to be documented in this file
 
+## [1.5.7] September 30, 2020
```
Comment (chukka) [inline stable/pipelines/CHANGELOG.md:4]: @rahulnlb1 Can you rebase your changes ?

Comment (rahulnlb1) [reply stable/pipelines/CHANGELOG.md]: Done

**Review by rimusz — APPROVED** [review]
LGTM

**Review by chukka — APPROVED** [review]

**Review by rimusz — APPROVED** [review]
LGTM

**Pull request merged**